lib/colorado_lottery.rb

```diff
@@ -0,0 +1,44 @@
+class ColoradoLottery
+
+  attr_reader :registered_contestants, :winners, :current_contestants
+
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants = {}
+  end
+
+  def interested_and_18?(contestant, game)
+    contestant.age >= 18 && contestant.game_interests.include?(game.name)
+  end
+
+  def can_register?(contestant, game)
+    interested_and_18?(contestant, game) && (contestant.out_of_state? == false || game.national_drawing? == true)
+  end
+
+  def register_contestant(contestant, game)
+    if can_register?(contestant, game)
+      if @registered_contestants[game.name] == nil
+        @registered_contestants[game.name] = [contestant]
+      else
+        @registered_contestants[game.name] << contestant
+      end
+    end
+  end
+
+  def eligible_contestants(game)
+    contestants_eligible = []
+    registered_contestants[game.name].each do |contestant|
```

Comment:
Nice job scoping to this to the `game.name` that was passed in. That's what we were hoping to see you do here. Remember with an each that we need to make sure that the data type we are building -- in this case `contestants_eligible` -- needs to be called at the end of this method, so that it becomes the return value.